src/main/java/com/hibernate/cinema/service/impl/UserServiceImpl.java

```diff
@@ -3,29 +3,37 @@
 import com.hibernate.cinema.dao.UserDao;
 import com.hibernate.cinema.model.User;
 import com.hibernate.cinema.service.UserService;
-import com.hibernate.cinema.util.HashUtil;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.stereotype.Service;
 
 @Service
 public class UserServiceImpl implements UserService {
     private final UserDao userDao;
+    private final PasswordEncoder encoder;
 
     @Autowired
-    public UserServiceImpl(UserDao userDao) {
+    public UserServiceImpl(UserDao userDao, PasswordEncoder encoder) {
         this.userDao = userDao;
+        this.encoder = encoder;
     }
 
     @Override
     public User add(User user) {
-        user.setSalt(HashUtil.getSalt());
-        String hashedPassword = HashUtil.hashPassword(user.getPassword(), user.getSalt());
-        user.setPassword(hashedPassword);
+        user.setPassword(encoder.encode(user.getPassword()));
         return userDao.add(user);
     }
 
     @Override
     public User findByEmail(String email) {
         return userDao.findByEmail(email).get();
     }
+
+    @Override
+    public User findUser(Authentication authentication) {
+        UserDetails userDetails = (UserDetails) authentication.getPrincipal();
```

Comment:
have you tried taking `authentication.getName()`?

Reply:
yes, I did, still don't have any clue how it works then